osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -109,7 +109,15 @@ public string ModsJson
         public string UserString
         {
             get => User?.Username;
-            set => User = new User { Username = value };
+            set => User = new User { Username = value, Id = UserID};
+        }
+
+        [JsonIgnore]
+        [Column("UserID")]
+        public long UserID
+        {
+            get => User?.Id ?? 1;
+            set => User = new User {Username = UserString, Id = value};
```

Comment:
probably want to check if a user is already existing rather than creating two user objects every deserialize (right now it's going to do so, due to UserString *and* UserID being deserialised).